### PR TITLE
fixed sapi dispatcher for namespace issues, and improper injection

### PR DESCRIPTION
**Pull request by james-nesbitt**
Fixed bad namespace for plugin manager, and really poor coding on the manager injection.

### CODE REVIEW

File: src/SapiDispatcher.php [hunk 1]
```diff
@@ -52,13 +54,13 @@ public function __construct(ConfigManager $configManager, AccountProxy $currentU
...
     $this->currentRouteMatch = $currentRouteMatch;
-    $this->statisticsPluginManager = \Drupal::service('plugin.manager.statisticsplugin');
+    $this->statisticsPluginManager = $statisticsPluginManager;
   }
 
    /**
     * Dispatches the statistics item to interested parties.
     *
-    * @param StatisticsItemInterface $item
+    * @param \Drupal\sapi\StatisticsItemInterface $item
```
Comment (james-nesbitt) [inline src/SapiDispatcher.php]: tehcnically this full namespace is not necessary.

**Comment by james-nesbitt** [comment]
@maijs what do you think about the use of the PluginManagerInterface in the dependency injection?  Should we restrict to the actual class, or create an interface?  or leave this?

File: src/SapiDispatcher.php [hunk 2]
```diff
@@ -5,6 +5,8 @@
 use Drupal\Core\Config\ConfigManager;
 use Drupal\Core\Session\AccountProxy;
 use Drupal\Core\Routing\CurrentRouteMatch;
+use Drupal\Component\Plugin\PluginManagerInterface;
+use Drupal\sapi\StatisticsItemInterface;
```
Comment (james-nesbitt) [inline src/SapiDispatcher.php]: @maijs is the full StatisticsItemInterface use statement necessary when the two elements are in the same namespace?

**Comment by maijs** [comment]
@james-nesbitt I prefer to expect interfaces in constructors. Take `EntityTypeManager`, it extends `DefaultPluginManager` (which implements `PluginManagerInterface`) and implements its own `EntityTypeManagerInterface` (which extends `PluginManagerInterface`). My idea is that when the scope of `StatisticsPluginManager` goes beyond `PluginManagerInterface` (custom methods to get plugins that only listen to particular event), when we can create a separate interface and use that.

**Comment by james-nesbitt** [comment]
@maijs agreed

**Comment by guncha25** [comment]
Looks better. Im fine with this.

**Pull request merged**